### DISCUSSION TITLE
reorder-list after multi-fileupload

### DISCUSSION THREAD

**Discussion by sbritz** [filamentphp/filament#1616] [Help]
Hi,
filament is awesome!! Great work!

I've got a little problem. I use the FileUpload:make() with multiple() and enableReordering().
Upload works, but how can I get the order-list after reordering with filepond?
Thx for your help :-)

**Comment by danharrin** [comment]
Hey! I think that the order is saved to the JSON in your database? As in, the array of URLs should be in the correct order? So the index of each URL is the order?

**Reply by sbritz** [reply]
Unfortunately not. I also thought that it works, but the order of the database is another. I don't know why..

**Reply by danharrin** [reply]
Okay, please open an issue if the behaviour is unexpected, with reproducible code

**Reply by MaartenJC** [reply]
I experience the same problem. How to reproduce:
Upload images where filenames start with 2 and 4.
Re-order the images so 4 comes before 2
<img width="481" alt="image" src="https://github.com/filamentphp/filament/assets/5759400/916db37e-e620-42d1-abf1-5fa2d99c69a6">

In handleRecordUpdate this is the dumped $data where the ordering is wrong. As you can see the filename with 2 comes first:

array:2 [▼ // app/Filament/Resources/ArticleImagesResource/Pages/EditArticleImages.php:40
  "images/products/org/29000/kOgD10X3WmKMpvUZbocG2yOPedSPzA-metaMmltZ181MTIweDI4ODBfM3gxNmJpdF9SR0JfY29sb3JfYmFyc19DTVlLV1JHQl8xMDBJUkUucG5n-.png" => "2img_5120x2880_3x16bit_RGB_color_bars_CMYKWRGB_100IRE.png"
  "images/products/org/29000/gfWN19586Go8YFsQeW4nwcEMw6PhrR-metaNGltZ181MTIweDI4ODBfM3gxNmJpdF9SR0JfY29sb3JfYmFyc19DTVlLV1JHQi5wbmc=-.png" => "4img_5120x2880_3x16bit_RGB_color_bars_CMYKWRGB.png"
]

**Reply by MaartenJC** [reply]
This is the FileUpload:
FileUpload::make('image')
                    ->label('Upload images')
                    ->disk('public')
                    ->storeFileNamesIn('image_file_names')
                    ->visibility('public')
                    ->maxSize(2048)
                    ->multiple()
                    ->enableReordering(),

**Reply by MaartenJC** [reply]
After some more testing and researching i found out that re-ordering works after the file upload is completed.
